lib/Graph/Nodes.cpp

```diff
@@ -574,6 +574,10 @@ void BatchedReduceAddNode::verify() const {
   assert(getBatch().dims().size() > 0 && "Invalid shape");
 }
 
+void LengthsSumNode::verify() const {
+  assert(getLengths().dims().size() == 1 && "Lengths must be a 1D vector");
```

Comment:
can you add this verification early in the loader?

Reply:
Why?
It shouldn't be in the loader then, it should be in `createLengthsSum()` method. But I'm still curious to understand what goes to this verifier, and what should be checked right away.

Reply:
Incorrect user input should be validated as soon as possible.
And in the case of ONNXIFI, we should report that we do not support this early on rather than failing on compilation/execution.